libunwind/src/libunwind.cpp

```diff
@@ -217,12 +223,24 @@ _LIBUNWIND_WEAK_ALIAS(__unw_get_proc_info, unw_get_proc_info)
 
 /// Resume execution at cursor position (aka longjump).
 _LIBUNWIND_HIDDEN int __unw_resume(unw_cursor_t *cursor) {
+  typedef LocalAddressSpace::pint_t pint_t;
   _LIBUNWIND_TRACE_API("__unw_resume(cursor=%p)", static_cast<void *>(cursor));
 #if __has_feature(address_sanitizer) || defined(__SANITIZE_ADDRESS__)
   // Inform the ASan runtime that now might be a good time to clean stuff up.
   __asan_handle_no_return();
 #endif
   AbstractUnwindCursor *co = (AbstractUnwindCursor *)cursor;
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(_LIBUNWIND_SANDBOX_OTYPES)
+  LocalAddressSpace &addressSpace = LocalAddressSpace::sThisAddressSpace;
+  pint_t sealer = addressSpace.getUnwindSealer();
+  if (addressSpace.isValidSealer(sealer)) {
+#ifdef _LIBUNWIND_SANDBOX_HARDENED
```

Comment:
Why not just add it to the existing #if? If this isn't defined we do a bunch of queries only to then do nothing with the result.